pkg/events/events.go

```diff
@@ -5034,6 +5028,12 @@ var Definitions = eventDefinitions{
 			Probes: []probeDependency{
 				{Handle: probes.SecurityMmapAddr, Required: true},
 				{Handle: probes.SecurityFileMProtect, Required: true},
+				{Handle: probes.SyscallEnter__Internal, Required: true},
+			},
+			Dependencies: dependencies{
+				TailCalls: []TailCall{
+					{MapName: "sys_enter_init_tail", MapIndexes: []uint32{uint32(Mmap), uint32(Mprotect)}, ProgName: "sys_enter_init"},
```

Comment:
So, apart from dynamically adding indexes for the syscalls being traced (done in initialization), we declare a "sys_enter_init" tailcall dependency to each event that REALLY DEPENDS on the syscall entry data (so if they are traced, but not their dependent syscall, we have the syscall tailcall as well).

And we can't depend on the syscall event (and add the tailcall there) because it would always include the tail call.. if I got this right.

Events that would depend on the init_tail are, for now: mem_prot_alert, most of, if not all, security events (file open, socket listen, socket connect, socket accept, socket bind, setsockopt, file mprotect)...

Reply:
You got it right exactly, think we should add this as a comment before merging, where do you think it would fit?

If we depend on the event it would include the submit tail call too (because it would mark the event for full dependency loading), that's why we add the tail call dependency directly, avoiding the syscall event.